test/e2e/e2e_suite_test.go

```diff
@@ -0,0 +1,314 @@
+	Expect(clusterctlConfigPath).To(BeAnExistingFile(), "Invalid argument. clusterctlConfigPath must be an existing file when calling initBootstrapCluster")
+	Expect(e2eConfig.GetVariable(operaratorPackage)).To(BeAnExistingFile(), "Invalid path to operator package. Please specify a valid one")
+	logFolder := filepath.Join(artifactFolder, "clusters", bootstrapClusterProxy.GetName())
+	Expect(os.MkdirAll(logFolder, 0750)).To(Succeed(), "Invalid argument. Log folder can't be created for initBootstrapCluster")
+
+	By("Adding docker variables secret")
+	Expect(bootstrapClusterProxy.Apply(ctx, dockerVariablesSecret)).To(Succeed())
+
+	By("Installing CAPI operator chart")
+	chart := &operatorframework.HelmChart{
+		BinaryPath:      helmBinaryPath,
+		Path:            e2eConfig.GetVariable(operaratorPackage),
+		Name:            "capi-operator",
+		Kubeconfig:      bootstrapClusterProxy.GetKubeconfigPath(),
+		Output:          operatorframework.Full,
+		AdditionalFlags: []string{"-n", operatorNamespace, "--create-namespace", "--wait"},
+	}
+	_, err := chart.InstallChart(map[string]string{
+		"cert-manager.enabled": "true",
+		"infrastructure":       "docker",
+		"secretName":           "variables",
+		"secretNamespace":      "default",
+	})
+	Expect(err).ToNot(HaveOccurred())
+
+	By("Installing rancher-turtles chart")
+	chart = &operatorframework.HelmChart{
+		BinaryPath:      helmBinaryPath,
+		Path:            chartPath,
+		Name:            "rancher-turtles",
+		Kubeconfig:      bootstrapClusterProxy.GetKubeconfigPath(),
+		Output:          operatorframework.Full,
+		AdditionalFlags: []string{"-n", rancherTurtlesNamespace, "--create-namespace", "--wait"},
+	}
+	_, err = chart.InstallChart(nil)
+	Expect(err).ToNot(HaveOccurred())
+
+	By("Installing rancher chart")
+	chart = &operatorframework.HelmChart{
+		BinaryPath: helmBinaryPath,
+		Path:       rancherPath,
+		Name:       "rancher",
+		Kubeconfig: bootstrapClusterProxy.GetKubeconfigPath(),
+		Output:     operatorframework.Full,
+		AdditionalFlags: []string{
+			"--version", e2eConfig.GetVariable(rancherVersion),
+			"--namespace", rancherNamespace,
+			"--create-namespace",
+			"--wait",
+		},
+	}
+	_, err = chart.InstallChart(map[string]string{
+		"bootstrapPassword":         "admin",
+		"features":                  e2eConfig.GetVariable(rancherFeatures),
+		"global.cattle.psp.enabled": "false",
+		"replicas":                  "1",
+		"hostname":                  e2eConfig.GetVariable(rancherHostname),
+	})
+	Expect(err).ToNot(HaveOccurred())
+}
+
+// Using a SynchronizedAfterSuite for controlling how to delete resources shared across ParallelNodes (~ginkgo threads).
+// The bootstrap cluster is shared across all the tests, so it should be deleted only after all ParallelNodes completes.
+var _ = SynchronizedAfterSuite(func() {
+	// After each ParallelNode.
+}, func() {
+	// After all ParallelNodes.
```

Comment:
we can dump artifacts here

Reply:
Added correct machine log handling here. Really helpful, thanks